ui/src/views/Clients/ViewClients/index.jsx

```diff
@@ -37,50 +42,36 @@ import ErrorPanel from '../../../components/ErrorPanel';
 export default class ViewClients extends PureComponent {
   state = {
     clientSearch: '',
-    // eslint-disable-next-line react/no-unused-state
-    previousClientId: '',
-    // This needs to be initially null in order for the defaultValue to work
-    value: null,
+    // This needs to be initially undefined in order for defaultValue to work
+    value: undefined,
   };
 
-  static getDerivedStateFromProps(props, state) {
-    // Any time the current user changes,
-    // Reset state to reflect new user / log out and default clientSearch query
-    if (
-      props.user &&
-      props.user.credentials.clientId !== state.previousClientId
-    ) {
-      return {
-        clientSearch: props.user.credentials.clientId,
-        previousClientId: props.user.credentials.clientId,
-      };
-    }
-
-    if (!props.user && state.previousClientId !== '') {
-      return {
-        clientSearch: '',
-        previousClientId: '',
-      };
-    }
-
-    return null;
-  }
-
-  handleClientSearchSubmit = clientSearch => {
+  handleClientSearchSubmit = async clientSearch => {
     const {
       data: { refetch },
     } = this.props;
+    const searchUri = this.props.history.location.search
+      ? decodeURIComponent(parse(this.props.history.location.search)['?search'])
+      : '';
 
     this.setState({ clientSearch });
 
-    refetch({
+    await refetch({
       clientOptions: {
         ...(clientSearch ? { prefix: clientSearch } : null),
       },
       clientsConnection: {
         limit: VIEW_CLIENTS_PAGE_SIZE,
       },
     });
+
+    if (clientSearch !== searchUri) {
+      this.props.history.push(
+        clientSearch.length > 0
+          ? `?search=${encodeURIComponent(clientSearch)}`
```

Comment:
You can probably use `stringify` instead. Something like

https://github.com/taskcluster/taskcluster/blob/0d6377a32c874be65929b36149d13eaed8dfd5d1/ui/src/views/PulseMessages/index.jsx#L156